### ISSUE TITLE
JSDT version conflict with Web Dev Tools

### ISSUE THREAD

**Issue by GoogleCodeExporter** [erhanselman/mobile-web-development-with-phonegap#5]
```
What steps will reproduce the problem?
1. Install fresh Eclipse 3.6 (I used C++ version)
2. Install PhonegapForAndroid 1.1 
3. Attempt install of Eclipse Web Developer Tools

What is the expected output? What do you see instead?
Expect to install Web Tools. Fails with:

Cannot complete the install because of a conflicting dependency.
  Software being installed: Eclipse Web Developer Tools 3.2.2.v201008100100-7O7CFbKEMf84nUlEJyX7-SPQjhYZ (org.eclipse.wst.web_ui.feature.feature.group 3.2.2.v201008100100-7O7CFbKEMf84nUlEJyX7-SPQjhYZ)
  Software currently installed: PhonegapForAndroid 1.1.0.201101042036 (com.mds.phonegapForAndroid.feature.group 1.1.0.201101042036)
  Only one of the following can be installed at once: 
    JavaScript Debug Core 1.0.1.v201008170101 (org.eclipse.wst.jsdt.debug.core 1.0.1.v201008170101)
    JavaScript Debug Core 1.0.3.v201011040143 (org.eclipse.wst.jsdt.debug.core 1.0.3.v201011040143)
    JavaScript Debug Core 1.0.0.v201006082045 (org.eclipse.wst.jsdt.debug.core 1.0.0.v201006082045)
  Cannot satisfy dependency:
    From: PhonegapForAndroid 1.1.0.201101042036 (com.mds.phonegapForAndroid.feature.group 1.1.0.201101042036)
    To: org.eclipse.wst.jsdt.feature.feature.group [1.2.3.v201010160635-7C78FGUF9JgLWLQ1UIPrac]
  Cannot satisfy dependency:
    From: JavaScript Development Tools 1.2.2.v201008170101-7C78FGMF9JgLWLMUUEMtXc (org.eclipse.wst.jsdt.feature.feature.group 1.2.2.v201008170101-7C78FGMF9JgLWLMUUEMtXc)
    To: org.eclipse.wst.jsdt.debug.core [1.0.1.v201008170101]
  Cannot satisfy dependency:
    From: JavaScript Development Tools 1.2.3.v201010160635-7C78FGUF9JgLWLQ1UIPrac (org.eclipse.wst.jsdt.feature.feature.group 1.2.3.v201010160635-7C78FGUF9JgLWLQ1UIPrac)
    To: org.eclipse.wst.jsdt.debug.core [1.0.3.v201011040143]
  Cannot satisfy dependency:
    From: Eclipse Web Developer Tools 3.2.2.v201008100100-7O7CFbKEMf84nUlEJyX7-SPQjhYZ (org.eclipse.wst.web_ui.feature.feature.group 3.2.2.v201008100100-7O7CFbKEMf84nUlEJyX7-SPQjhYZ)
    To: org.eclipse.wst.jsdt.feature.feature.group [1.2.2.v201008170101-7C78FGMF9JgLWLMUUEMtXc]

What version of the product are you using? On what operating system?
PhoneGap plugin 1.1; Eclipse 3.6

Please provide any additional information below.
Can the PhoneGap plugin JSDT dependency be widened, so it can coexist with 
other JS-centric tooling?

```

Original issue reported on code.google.com by `wa...@stidolph.com` on 8 Jan 2011 at 5:03

**Comment by GoogleCodeExporter** [comment]
```
The JSDT components included with the PhoneGap plugin are from Web Developer 
Tools 3.2.3 from http://download.eclipse.org/webtools/downloads/

The JSDT components included in WTP 3.2.2 have problems with typical mobile web 
UI framework libraries.

Does it work for you to install WTP 3.2.3?
```

Original comment by `paul.beu...@gmail.com` on 8 Jan 2011 at 10:48

**Comment by GoogleCodeExporter** [comment]
```
It looks like I can use 3.2.3, but not 3.2.2 nor 3.3.0M4 ... fyi, with
3.3.0M4  I get:

Cannot complete the install because of a conflicting dependency.
  Software being installed: Eclipse Web Developer Tools
3.3.0.v201010270400-7O7IFhJEMi9Cr0KHBrlhE5UuG-cH
(org.eclipse.wst.web_ui.feature.feature.group
3.3.0.v201010270400-7O7IFhJEMi9Cr0KHBrlhE5UuG-cH)
  Software currently installed: PhonegapForAndroid 1.1.0.201101042036
(com.mds.phonegapForAndroid.feature.group 1.1.0.201101042036)
  Only one of the following can be installed at once:
    JavaScript Debug Core 2.0.100.v201011300730
(org.eclipse.wst.jsdt.debug.core 2.0.100.v201011300730)
    JavaScript Debug Core 1.0.1.v201008170101
(org.eclipse.wst.jsdt.debug.core 1.0.1.v201008170101)
    JavaScript Debug Core 1.0.3.v201011040143
(org.eclipse.wst.jsdt.debug.core 1.0.3.v201011040143)
    JavaScript Debug Core 1.0.0.v201006082045
(org.eclipse.wst.jsdt.debug.core 1.0.0.v201006082045)
  Cannot satisfy dependency:
    From: PhonegapForAndroid 1.1.0.201101042036
(com.mds.phonegapForAndroid.feature.group 1.1.0.201101042036)
    To: org.eclipse.wst.jsdt.feature.feature.group
[1.2.3.v201010160635-7C78FGUF9JgLWLQ1UIPrac]
  Cannot satisfy dependency:
    From: JavaScript Development Tools
1.2.3.v201010160635-7C78FGUF9JgLWLQ1UIPrac
(org.eclipse.wst.jsdt.feature.feature.group
1.2.3.v201010160635-7C78FGUF9JgLWLQ1UIPrac)
    To: org.eclipse.wst.jsdt.debug.core [1.0.3.v201011040143]
  Cannot satisfy dependency:
    From: JavaScript Development Tools
1.3.0.v201008190258-7F78FUHFBBoPbPY-lyXBmc
(org.eclipse.wst.jsdt.feature.feature.group
1.3.0.v201008190258-7F78FUHFBBoPbPY-lyXBmc)
    To: org.eclipse.wst.jsdt.debug.core [2.0.100.v201011300730]
  Cannot satisfy dependency:
    From: Eclipse Web Developer Tools
3.3.0.v201010270400-7O7IFhJEMi9Cr0KHBrlhE5UuG-cH
(org.eclipse.wst.web_ui.feature.feature.group
3.3.0.v201010270400-7O7IFhJEMi9Cr0KHBrlhE5UuG-cH)
    To: org.eclipse.wst.jsdt.feature.feature.group
[1.3.0.v201008190258-7F78FUHFBBoPbPY-lyXBmc]On Sat, Jan 8, 2011 at 2:49 PM,
<mobile-web-development-with-phonegap.eclipselabs.org@codespot.com> wrote:
```

Original comment by `wa...@stidolph.com` on 8 Jan 2011 at 11:24

**Comment by GoogleCodeExporter** [comment]
```
Thanks for the update.

Does 3.2.3 work for you?  Does 3.3.0M4 have something compelling?

In any case, I'll investigate if I can bundle JSDT so that it can be separably 
updateable from the Phonegap plugin - perhaps bundle in the site instead of the 
feature.

```

Original comment by `paul.beu...@gmail.com` on 8 Jan 2011 at 11:34
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
Original comment by `paul.beu...@gmail.com` on 8 Jan 2011 at 11:34

**Comment by GoogleCodeExporter** [comment]
```
No, nothing compelling to me in 3.3 (I'm pretty much a newbie at web dev, so I 
don't use any advanced features!); I only checked that to see if the issue was 
the plugin being tied to a specific single release or if it was "any WTP at or 
after 3.2.2"
```

Original comment by `wa...@stidolph.com` on 8 Jan 2011 at 11:46

**Comment by GoogleCodeExporter** [comment]
```
Thanks Wayne.

Good find.  It does need to be fixed to work with 3.2.3 and later.
```

Original comment by `paul.beu...@gmail.com` on 8 Jan 2011 at 11:48

**Comment by GoogleCodeExporter** [comment]
```
1.1.1 puts the JSDT feature and HTML feature into the Eclipse site instead of 
making them dependencies for the PhoneGap feature itself.

This enables JSDT and HTML to be updated separately from the PhoneGap plugin

Unfortunately, it requires an update from 1.1.0 to 1.1.1 to be done twice since 
the first update uninstalls JSDT and HTML and only the second will install them.
```

Original comment by `paul.beu...@gmail.com` on 12 Jan 2011 at 7:49
- Changed state: **Fixed**
